src/main/java/com/github/rnewson/couchdb/lucene/util/Analyzers.java

```diff
@@ -186,13 +280,313 @@ protected TokenStreamComponents wrapComponents(String fieldName, TokenStreamComp
+			}
+		}
+
+		return parameters;
+	}
+
+	/**
+	 * Retrieve configuration information from one json param object. Type
+	 * information is used to construct actual data containing objects.
+	 * 
+	 * Each param object looks like:
+	 * 
+	 * <pre>{ "name": &lt;a name>, "type": &lt;oneof: set, bool, int, file, string>, "value": &lt;value> }</pre>
+	 * 
+	 * The name is not used. Values of type <code>set</code> are JSON arrays and 
+	 * are used to represent lucene CharArraySets such as for stop words in StandardAnalyzer
+	 *
+	 * @param param Element that represents <param/>
+	 * @return Triple key-value-value-type
+	 * @throws org.exist.indexing.lucene.AnalyzerConfig.ParameterException
+	 */
+	private static KeyTypedValue getConstructorParameter(JSONObject param) throws ParameterException, JSONException {
+
+		final String name = param.optString("name");
+		final String type = param.optString("type", "string");
+		final String value = param.optString("value");
+
+		KeyTypedValue parameter = null;
+
+		switch (type) {
+
+		case "string": {
+			if (value == null) {
+				throw new ParameterException("The 'value' field of a string param must exist and must contain a String value.");
+			}
+
+			parameter = new KeyTypedValue(name, value, String.class);
+
+			break;
+		}
+
+		// "java.io.FileReader":
+		case "file": {
+
+			if (value == null) {
+				throw new ParameterException("The 'value' field of a file param must exist and must contain a file name.");
+			}
+
+			try {
+				// ToDo: check where to close reader to prevent resource leakage
+				Reader fileReader = new java.io.FileReader(value);
+				parameter = new KeyTypedValue(name, fileReader, Reader.class);
+
+			} catch (java.io.FileNotFoundException ex) {
+				logger.error(String.format("File '%s' could not be found.", value), ex);
+			}
+			break;
+		}
+
+		// "org.apache.lucene.analysis.util.CharArraySet":
+		case "set": {
+			JSONArray values = param.optJSONArray("value");
+
+			if (values == null) {
+				throw new ParameterException("The 'value' field of a set param must exist and must contain a json array of strings.");
+			}
+
+			final Set<String> set = new HashSet<>();
+
+			for (int i = 0; i < values.length(); i++) {
+				set.add(values.getString(i));
+			}
+
+			parameter = new KeyTypedValue(name, CharArraySet.copy(set), CharArraySet.class);
+			break;
+		}
+
+		// "java.lang.Integer":
+		case "int":
+
+			int n = param.optInt("value");
+			parameter = new KeyTypedValue(name, n, int.class);
+
+			break;
+
+		// "java.lang.Boolean":
+		case "boolean":
+
+			boolean b = param.optBoolean("value");
+			parameter = new KeyTypedValue(name, b, boolean.class);
+			break;
+
+		default:
+			// there was no match
+			logger.error(String.format("Unknown lucene analyzer parameter type: %s", type));
+			break;
+		}
+
+		return parameter;
+	}
+
+	/**
+	 * CLass for containing the Triple : key (name), corresponding value and
+	 * class type of value.
+	 */
+	private static class KeyTypedValue {
```

Comment:
Thank you for your very thorough consideration of the pull request. I've created a new branch in my fork and made 3 clean commits: 1) implementing the analyzer configuration; 2) adding tests; and 3) adding documentation. I tried to answer all of the comments in the new branch. I was thinking to close this PR and make a new one with the new "clean" branch. I ended up making far too many changes to be comfortable trying to do everything with fixup and so on. Let me know if this approach will work for you.